fuel/datasets/hdf5.py

```diff
@@ -64,3 +67,127 @@ def get_data(self, state=None, request=None):
                 raise ValueError
         data = [node[request] for node in self.nodes]
         return data
+
+
+@do_not_pickle_attributes('data_sources')
+class H5PYDataset(Dataset):
+    """An h5py-fueled HDF5 dataset.
+
+    This dataset class assumes a particular file layout:
+
+    * Data sources reside in the root group, and their names define the
+      source names.
+    * The dataset is not explicitly split. Instead, splits are defined as
+      attributes of the root group. They're expected to be numpy arrays of
+      shape (2,), with the first element being the starting point
+      (inclusive) of the split and the last element being the stopping
+      point (exclusive) of the split.
+
+    The `which_set`, `start` and `stop` parameters work together in the
+    following way:
+
+    * `which_set` is resolved first. If it is `None`, the whole dataset is
+      used.
+    * `start` and `stop` define a slice *within the context of*
```

Comment:
I've been thinking of proposing that we start using range objects instead of 'start' and 'stop', since a) one less function parameter and b) allows for the possibility of taking step > 1. (Even further, we could one day accept either a range object or a list of indices, much as if you were indexing a NumPy array.) @bartvm @vdumoulin ?